routing/base/followed_polyline.hpp

```diff
@@ -115,6 +122,48 @@ class FollowedPolyline
     return res;
   }
 
+  /// \returns pair of iterator (projection point) and bool (true if nearest point is on unmatched segment)
+  template <typename DistanceFn>
+  std::pair<Iter, bool> GetClosestMatchedProjectionInInterval(m2::RectD const & posRect, DistanceFn const & distFn,
+                                                              size_t startIdx, size_t endIdx) const
+  {
+    CHECK_LESS_OR_EQUAL(endIdx, m_segProj.size(), ());
+    CHECK_LESS_OR_EQUAL(startIdx, endIdx, ());
+    Iter nearestIter;
+    double minDist = std::numeric_limits<double>::max();
+    double minDistUnmatched = minDist;
+
+    m2::PointD const currPos = posRect.Center();
+
+    for (size_t i = startIdx; i < endIdx; ++i)
+    {
+      m2::PointD const & pt = m_segProj[i].ClosestPointTo(currPos);
+
+      if (!posRect.IsPointInside(pt))
+        continue;
+
+      Iter it(pt, i);
+      double const dp = distFn(it);
+      if(dp >= minDistUnmatched && dp >= minDist)
+        continue;
+
+      if (std::find(m_unmatchedSegmentIndexes.begin(), m_unmatchedSegmentIndexes.end(), it.m_ind) == m_unmatchedSegmentIndexes.end())
```

Comment:
`if (binary_search(m_unmatchedSegmentIndexes.begin(), m_unmatchedSegmentIndexes.end(), it.m_ind)`?